modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceProductCard.testcase

```diff
@@ -1064,160 +1064,210 @@ definition {
 
-		Select(
-			key_fieldLabel = "Price Type",
-			locator1 = "Select#GENERIC_SELECT_FIELD",
-			value1 = "Static");
+			Select(
+				key_fieldLabel = "Price Type",
+				locator1 = "Select#GENERIC_SELECT_FIELD",
+				value1 = "Static");
 
-		Button.clickSave();
+			Button.clickSave();
 
-		Alert.viewSuccessMessage();
+			Alert.viewSuccessMessage();
 
-		Click(
-			key_index = "Black",
-			locator1 = "WorkflowMetrics#SEARCH_FIELD_MAPPINGS_METRICS_INDEXES");
+			Click(
+				key_index = "Black",
+				locator1 = "WorkflowMetrics#SEARCH_FIELD_MAPPINGS_METRICS_INDEXES");
 
-		Type(
-			key_fieldLabel = "Delta Price",
-			locator1 = "TextInput#GENERIC_TEXT_INPUT",
-			value1 = "60.00");
+			Type(
+				key_fieldLabel = "Delta Price",
+				locator1 = "TextInput#GENERIC_TEXT_INPUT",
+				value1 = "60.00");
 
-		Type(
-			key_fieldLabel = "SKU",
-			locator1 = "TextInput#GENERIC_TEXT_INPUT",
-			value1 = "MIN55861");
+			Type(
+				key_fieldLabel = "SKU",
+				locator1 = "TextInput#GENERIC_TEXT_INPUT",
+				value1 = "MIN55861");
 
-		Button.clickSave();
+			Button.clickSave();
 
-		Alert.viewSuccessMessage();
+			Alert.viewSuccessMessage();
 
-		Refresh();
+			Refresh();
+		}
 
-		CommerceEntry.gotoMenuTab(menuTab = "SKUs");
+		task ("and an SKU is created and made Purchasable") {
+			CommerceEntry.gotoMenuTab(menuTab = "SKUs");
 
-		Click(locator1 = "Button#PLUS");
+			Click(locator1 = "Button#PLUS");
 
-		Click(
-			key_fieldName = "Add SKU",
-			locator1 = "FormViewBuilder#OBJECT_FIELD_BUTTON_DROPDOWN");
+			Click(
+				key_fieldName = "Add SKU",
+				locator1 = "FormViewBuilder#OBJECT_FIELD_BUTTON_DROPDOWN");
 
-		SelectFrame.selectFrameNoLoading(locator1 = "CommerceEntry#IFRAME_SIDE_PANEL");
+			SelectFrame.selectFrameNoLoading(locator1 = "CommerceEntry#IFRAME_SIDE_PANEL");
 
-		Type(
-			key_fieldLabel = "SKU",
-			locator1 = "TextInput#GENERIC_TEXT_INPUT",
-			value1 = "BLACKSKU");
+			Type(
+				key_fieldLabel = "SKU",
+				locator1 = "TextInput#GENERIC_TEXT_INPUT",
+				value1 = "BLACKSKU");
 
-		Click(locator1 = "Dropdown#TRIGGER");
+			Click(locator1 = "Dropdown#TRIGGER");
 
-		Click(
-			key_fieldName = "Black",
-			locator1 = "FormViewBuilder#OBJECT_FIELD_BUTTON_DROPDOWN");
+			Click(
+				key_fieldName = "Black",
+				locator1 = "FormViewBuilder#OBJECT_FIELD_BUTTON_DROPDOWN");
 
-		Check.checkToggleSwitch(
-			key_toggleSwitchLabel = "Purchasable",
-			locator1 = "ToggleSwitch#ANY_TOGGLE_SWITCH");
+			Check.checkToggleSwitch(
+				key_toggleSwitchLabel = "Purchasable",
+				locator1 = "ToggleSwitch#ANY_TOGGLE_SWITCH");
 
-		Type(
-			key_fieldLabel = "Base Price",
-			locator1 = "TextInput#GENERIC_TEXT_INPUT",
-			value1 = "10.0");
+			Type(
+				key_fieldLabel = "Base Price",
+				locator1 = "TextInput#GENERIC_TEXT_INPUT",
+				value1 = "10.0");
 
-		Button.click(button = "Publish");
+			Button.click(button = "Publish");
 
-		WaitForElementNotPresent(locator1 = "Message#SUCCESS");
+			WaitForElementNotPresent(locator1 = "Message#SUCCESS");
```

Comment:
Also looks like a single goal, might be better off as a macro

Reply:
Created CommerceOptions.enableUseInFacetedNavigation, CommerceOptions.enableRequired, and CommerceOptions.enableSKUContributor macros.